core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -1069,8 +1070,8 @@ protected BNode parseNodeID() throws IOException, RDFParseException {
 			reportError("Expected a letter, found '" + (char) c + "'", BasicParserSettings.PRESERVE_BNODE_IDS);
 		}
 
-		StringBuilder name = new StringBuilder(32);
-		name.append(Character.toChars(c));
```

Comment:
The other use of name.append below could also be modified, as its casting straight to a char may not be correct in all cases (I can't remember now if there are only BMP characters allowed for blank node label characters, but safer to reuse the Chars.append method instead of a risky cast.